Src/SmtpServer/SmtpSessionManager.cs

```diff
@@ -118,6 +102,20 @@ public SmtpSessionHandle(SmtpSession session, SmtpSessionContext sessionContext)
             public SmtpSessionContext SessionContext { get; }
 
             public Task CompletionTask { get; set; }
+
+            public override bool Equals(object obj)
```

Comment:
I dont think this will be needed, ie, there shouldnt be any reason to perform an equality chedck on the actual handle instance itself?

Reply:
You're right, that still comes from the HashSet variant I tested